crates/orchestrator/src/workers/snos.rs

```diff
@@ -8,7 +14,35 @@ impl Worker for SnosWorker {
     /// 1. Fetch the latest completed block from the Starknet chain
     /// 2. Fetch the last block that had a SNOS job run.
     /// 3. Create SNOS run jobs for all the remaining blocks
+    // TEST : added config temporarily to test
     async fn run_worker(&self) {
-        todo!()
+        let config = config().await;
+        let provider = config.starknet_client();
+        let latest_block_number = provider.block_number().await.expect("Failed to fetch block number from rpc");
+        let latest_block_processed_data = config
```

Comment:
Could it be the case that there are gaps in the snos job executions?
For instance:

block 1: snos job succeeded
block 2: snos job failed
block 3: snos job succeeded
block 4: snos job pending

Ideally you should also search for failed ones and retry (if possible)

Reply:
the job execution already has a retry logic. if something moved to a failed state, we have ideally already retried enough number of times and we should throw and alert and investigate manually, wdyt?

Reply:
Oh right, that makes sense 👍